nuxeo-services/nuxeo-platform-web-common/src/main/resources/web/nuxeo.war/login.jsp

```diff
@@ -526,35 +539,40 @@ a.mobileAppLink:hover {
     <fmt:message bundle="${messages}" key="label.mobile.openInApp" />
   </a>
 </div>
-<% } %>
 
 <script type="text/javascript">
-  <% if (displayMobileBanner) { %>
   // Build mobile app links if the mobile banner is displayed
-  var indexOfHash = window.location.href.indexOf('#!');
-  if (indexOfHash > -1) {
-    // lastPart = #!/doc/default/f6fa9686-3618-47a8-9419-ff1cc76fc857
-    // or lastPart = #!/doc/f6fa9686-3618-47a8-9419-ff1cc76fc857
-    var lastPart = window.location.href.substring(indexOfHash);
-
-    var docPart;
-    var parts = lastPart.split('/');
+  <% if (requestedUrlFragment != null) { %>
```

Comment:
Could also check if `requestedUrlFragment` is not blank since the cookie is always set no matter if the requested URL contains a hash or not, so for a JSF URL, the cookie value is `""`.